python/ray/train/tests/test_trainer.py

```diff
@@ -1,1301 +0,0 @@
-
-    def get_dataset():
-        pipeline_iterator = train.get_dataset_shard().iter_epochs()
-        data_all_epochs = []
-        for _ in range(2):
-            dataset_this_epoch = next(pipeline_iterator)
-            data_this_epoch = []
-            for batch in dataset_this_epoch.iter_batches(batch_format="default"):
-                data_this_epoch.extend(batch)
-
-            if len(data_all_epochs) > 0:
-                # Make sure data is shuffled per epoch.
-                assert data_this_epoch != data_all_epochs[-1]
-
-            data_all_epochs.append(data_this_epoch)
-        return data_all_epochs
-
-    config = TestConfig()
-
-    trainer = Trainer(config, num_workers=2)
-    trainer.start()
-    results = trainer.run(get_dataset, dataset=dataset)
-    check_dataset_output(num_data, num_epochs, results)
-
-
-def test_dataset_fault_tolerance(ray_start_4_cpus):
-    dataset = ray.data.range(10)
-    test_config = TestConfig()
-
-    def train_func():
-        return train.get_dataset_shard()
-
-    def train_actor_failure():
-        import sys
-
-        sys.exit(1)
-
-    new_backend_executor_cls = gen_new_backend_executor(train_actor_failure)
-
-    class SingleGetDatasetShardsBackendExecutor(new_backend_executor_cls):
-        def __init__(self, *args, **kwargs):
-            super().__init__(*args, **kwargs)
-            self._has_called_get_dataset_shards = False
-
-        def _get_dataset_shards(self, dataset_or_dict):
-            if self._has_called_get_dataset_shards:
-                raise Exception
-            self._has_called_get_dataset_shards = True
-            return super()._get_dataset_shards(dataset_or_dict)
-
-    with patch.object(
-        ray.train.trainer, "BackendExecutor", SingleGetDatasetShardsBackendExecutor
-    ):
-        trainer = Trainer(test_config, num_workers=2)
-        trainer.start()
-        trainer.run(train_func, dataset=dataset)
-        # No exception is raised by _get_dataset_shards
-
-
-@pytest.mark.parametrize("resource", ["CPU", "GPU", "extra"])
-@pytest.mark.parametrize("num_requested", [0.5, 1, 2])
-def test_resources(ray_start_4_cpus_4_gpus_4_extra, resource, num_requested):
-    num_workers = 2
-    config = TestConfig()
-    original = ray.available_resources().get(resource)
-    resources_per_worker = {resource: num_requested}
-    use_gpu = resource == "GPU"
-    trainer = Trainer(
-        config,
-        num_workers=num_workers,
-        use_gpu=use_gpu,
-        resources_per_worker=resources_per_worker,
-    )
-
-    trainer.start()
-    expected = original - num_workers * num_requested
-    wait_for_condition(lambda: ray.available_resources().get(resource, 0) == expected)
-
-    trainer.shutdown()
-    wait_for_condition(lambda: ray.available_resources().get(resource, 0) == original)
-
-    # Check that user input has not been modified
-    assert resources_per_worker == {resource: num_requested}
-
-
-def test_gpu_requests(ray_start_4_cpus_4_gpus_4_extra):
```

Comment:
This is an important test to keep...can we rewrite using DataParallelTrainer?

Reply:
We are already testing for that in `test_torch_get_device` and `test_torch_get_device_dist` in `test_gpu`.

Reply:
Added this back